### PR TITLE
Add link to catalog on hover

### PR DESCRIPTION
**Pull request by cubetastic33**
NP-17

Adds a link to the UTD catalog from the course description tooltip in the semester view and sidebar.

### CODE REVIEW

File: src/components/planner/CourseInfoHoverCard.tsx [hunk 1]
```diff
@@ -27,7 +30,17 @@ export const CourseInfoHoverCard: FC<CourseInfoHoverCardProps> = ({
         className="z-[999] w-[300px] animate-[slideUpAndFade_0.3s] rounded-md border border-neutral-200 bg-generic-white p-6 shadow-sm"
         sideOffset={5}
       >
-        <h3 className="mb-2 text-base font-semibold">{title}</h3>
+        <h3 className="mb-2 text-base font-semibold">
+          <Link
+            href={`https://catalog.utdallas.edu/2022/undergraduate/courses/${courseCode
```
Comment (thekevinge) [inline src/components/planner/CourseInfoHoverCard.tsx:35]: Everything looks good! 

Just one nitpick:
- Could we make the year dependent on the year of their first semester? You can use the ``SemesterContext`` or prop drill it.

**Review by akevinge — CHANGES_REQUESTED** [review]

File: src/components/planner/CourseInfoHoverCard.tsx (hunk 1 above)
Comment (cubetastic33) [reply src/components/planner/CourseInfoHoverCard.tsx:35]: Only concern I have with that is for example if someone started in Spring 2023 - There is no catalog yet for 2023. Should we just use their first semester capping it at the last Fall?

Comment (thekevinge) [reply src/components/planner/CourseInfoHoverCard.tsx:35]: Oh shoot that's a good point. From what I understand, which catalog you use is based on the Fall semester, so if you started Spring 2023 you still use the "2022" catalog. It kinda implies it [here](https://catalog.utdallas.edu/2022/undergraduate/home) too. For people that start in Fall 2023... we'll have to hope that the uni posts it, and if not we'll deal with it later.

**Review by akevinge — APPROVED** [review]
Looks good! Will merge at next meeting :+1:

**Pull request merged**